### ISSUE TITLE
Error calling NCO command with tempfile helper

### ISSUE THREAD

**Issue by monocongo** [nco/pynco#37]
I am attempting to use the Python NCO bindings for calling an NCO command using the [tempfile helper feature](https://pynco.readthedocs.io/en/latest/#tempfile-helpers). For example:

```
nco = Nco()
temp_netcdf_file = nco.ncpdq(input=arguments.netcdf_precip,
                             options=["-a \"-lat,lon,time\""])
```
My expectation from the above is that I will receive a string value with the name of the temporary NetCDF file created as a result of the NCO command. (Or maybe it will be a file object? The return type for this usage is unclear from the documentation.)

When I run the above I get an error:

```nco.nco.NCOException: (returncode:1) b''```

When I step through the `nco.py` code I see the command appears to be properly parsed:  

`cmd: <class 'list'>: ['/home/james/miniconda3/envs/climate/bin/ncpdq', '-a', '"-lat,lon,time"']`

However, when the command is called (line 263) it gets unexpected return values, resulting in the above-mentioned exception.

I can run the command in a terminal on Linux (Ubuntu 18.04) without issue, although in that case I am specifying an output file.

I have seen [similar behavior](https://stackoverflow.com/questions/48627509/pynco-error-when-issuing-ncks-command-unexpected-bytes-object-instead-of-strin) the previous time I attempted to use this package, perhaps these two issues are related?

Thanks in advance for any help with this.

**Comment by jhamman** [comment]
I think this usage should be returning a path to a temporary. I'm not actively working on this package but would be happy to help others fix any current problems.

**Comment by monocongo** [comment]
Thanks, @jhamman . There are several failing tests that I have fixed, plus a few more cleanups (PEP8, unused variables, etc.) that I've worked in. I will submit a PR soon. I appreciate your help.

**Comment by monocongo** [comment]
BTW while testing this with a Python2.7 virtual environment I am seeing some cases where the NCO executable path is being set from the OS environment's PATH, and in that case it doesn't find the NCO executables installed in the virtual environment but rather from the base OS location instead. This is problematic if NCO is installed into the OS but `ncdump` is not, i.e. when the NCO path is set it uses the location where `ncks` is located in the OS, such as `/usr/bin`, and when the test for `ncdump -h` is run it fails since `ncdump` isn't located under `/usr/bin` (file not found error). Maybe there's something amiss with how I have my `PATH` being set when running these tests under a 2.7 (Anaconda) virtual environment? I've tried this again using the previous virtual environment I was using before and seeing the same behavior. I'm not sure why this was working well before, i.e. it was locating the `ncks` in the virtual environment before the OS location and using the virtual environment's path to `ncks` as the NCO path, and since this matched with where ncdump was installed in the venv all was good. This behavior has changed since creating a new Python2.7 virtualenv using conda, and I assume that this has goobered up some PATH settings that were causing the venv path to be used, which I assume is what we want? I think the underlying issue for the test at play (`test_operator_prints_out()`) is that it tests `ncdump` using the path to `ncks`, and the two could be installed in separate locations since `ncdump` isn't part of NCO per se. Or am I missing something obvious?

TL;DR: The 2.7 support is a bit hosed, while trying to fix that I have run into issues with finding NCO path under the new 2.7 conda environment I've created for this. One test involving ncdump is failing since the path to ncdump isn't always the same as the path to ncks, especially in cases when NCO and/or ncdump is installed only within a virtual environment and not on the base OS.
